packages/flutter_tools/test/integration.shard/isolated/dart_data_asset_test.dart

```diff
@@ -0,0 +1,430 @@
+
+        final ProcessTestResult result = await runFlutter(
+          <String>['build', '-v', target],
+          root.path,
+          <Transition>[Barrier.contains('Built build/$target')],
+        );
+        if (result.exitCode != 0) {
+          throw Exception(
+            'flutter build failed: ${result.exitCode}\n${result.stderr}\n${result.stdout}',
+          );
+        }
+        final Directory buildTargetDir = root.childDirectory('build').childDirectory(target);
+
+        final List<File> manifestFiles =
+            buildTargetDir
+                .listSync(recursive: true)
+                .whereType<File>()
+                .where((File file) => file.path.endsWith('AssetManifest.json'))
+                .toList();
+
+        if (manifestFiles.isEmpty) {
+          throw Exception('Expected a `AssetManifest.json` to be avilable in the $buildTargetDir.');
+        }
+        for (final File manifestFile in manifestFiles) {
+          final Map<String, Object?> manifest =
+              json.decode(manifestFile.readAsStringSync()) as Map<String, Object?>;
+          for (final String id in available) {
+            final String key = 'packages/$packageName/$id';
+            final List<Object?> entry = manifest[key]! as List<Object?>;
+            expect(entry, equals(<String>[key]));
+
+            final File file = manifestFile.parent.childFile(key);
+            expect(file.readAsStringSync(), assets[id]);
+          }
+        }
+      });
+    }
+
+    for (final String target in <String>[hostOs, 'web']) {
+      testWithoutContext('flutter build $target with conflicting assets', () async {
+        final Map<String, String> assets = <String, String>{
+          'id1.txt': 'content1',
+          'id2.txt': 'content2',
+        };
+        final List<String> available = <String>['id1.txt'];
+        writeAssets(assets, root);
+        writeAssets(assets, rootDependency);
+        writeHookLibrary(root, assets, available: available);
+        writeHookLibrary(rootDependency, assets, available: available);
+        writeHelperLibrary(root, 'version1', assets.keys.toList());
+
+        await modifyPubspec(root, (YamlEditor editor) {
+          editor.update(
+            <String>['dependencies', packageNameDependency],
+            <String, String>{'path': '../$packageNameDependency'},
+          );
+        });
+
+        await modifyPubspec(rootDependency, (YamlEditor editor) {
+          editor
+            ..update(<String>['flutter', 'assets'], <String>[assets.keys.first])
+            ..update(<String>['dependencies'], <String, String>{'native_assets_cli': '^0.11.0'});
+        });
+
+        final ProcessTestResult result = await runFlutter(
+          <String>['build', '-v', target],
+          root.path,
+          <Transition>[
+            Barrier.contains(
+              'Conflicting assets: The asset "asset: packages/data_asset_dependency/id1.txt" was declared in the pubspec and the hook',
```

Comment:
Can logic such as this be tested on the general shard (unit tests that mock running the hooks)? That requires less resources to run.

See test/general.shard/isolated/native_assets_test.dart with `buildRunner: FakeFlutterNativeAssetsBuildRunner(`.

I'll leave the balance between mocked-out-tests and full integration tests is something for Ben to decide on.